streamer/src/nonblocking/quic.rs

```diff
@@ -702,17 +729,28 @@ async fn handle_connection(
     );
     let stable_id = connection.stable_id();
     stats.total_connections.fetch_add(1, Ordering::Relaxed);
+    let max_streams_per_100ms =
+        max_streams_for_connection_in_100ms(peer_type, params.stake, params.total_stake);
+    let mut last_throttling_instant = tokio::time::Instant::now();
+    let mut streams_in_current_interval = 0;
     while !stream_exit.load(Ordering::Relaxed) {
         if let Ok(stream) =
             tokio::time::timeout(WAIT_FOR_STREAM_TIMEOUT, connection.accept_uni()).await
         {
             match stream {
                 Ok(mut stream) => {
+                    if reset_throttling_params_if_needed(&mut last_throttling_instant) {
+                        streams_in_current_interval = 0;
+                    } else if streams_in_current_interval >= max_streams_per_100ms {
+                        let _ = stream.stop(VarInt::from_u32(STREAM_STOP_CODE_THROTTLING));
+                        continue;
+                    }
+                    streams_in_current_interval = streams_in_current_interval.saturating_add(1);
```

Comment:
this can be encapsulated as something like...

```rust
struct StreamThrottle {
    max_streams_per_interval: u64,
    interval_start: Instant,
    stream_count: usize,
}

impl StreamThrottle {
    pub fn new(max_stream_per_interval: u64) -> Self {
        Self {
            max_streams_per_interval,
            interval_start: tokio::time::Instant::now(),
            stream_count: 0,
        }
    }

    fn reset_throttling_params_if_needed(&mut self) -> bool {
        let now = tokio::time::Instant::now();
        let reset = now.duration_since(self.last_instant) > STREAM_THROTTLING_INTERVAL;
        if reset {
            self.last_instant = now;
        }
        reset
    }

    pub fn can_add_stream(&mut self) -> bool {
        if self.reset_throttling_params_if_needed() {
            self.stream_count = 0;
        } else if stream_count >= max_streams_per_interval {
            return false;

        }
        streams_in_current_interval = streams_in_current_interval.saturating_add(1);
        true
    }
}
```

then this block becomes

```diff
+    let mut stream_throttle = StreamThrottle::new(max_streams_per_100ms);
     while !stream_exit.load(Ordering::Relaxed) {
         if let Ok(stream) =
             tokio::time::timeout(WAIT_FOR_STREAM_TIMEOUT, connection.accept_uni()).await
         {
             match stream {
                 Ok(mut stream) => {
+                    if !stream_throttle.can_add_stream() {
+                        let _ = stream.stop(VarInt::from_u32(STREAM_STOP_CODE_THROTTLING));
+                        continue;
+                    }